Svc/FileManager/FileManager.cpp

```diff
@@ -156,10 +164,106 @@ void FileManager ::FileSize_cmdHandler(const FwOpcodeType opCode, const U32 cmdS
     this->sendCommandResponse(opCode, cmdSeq, status);
 }
 
+void FileManager ::ListDirectory_cmdHandler(const FwOpcodeType opCode, const U32 cmdSeq, const Fw::CmdStringArg& dirName) {
+    Fw::LogStringArg logStringDirName(dirName.toChar());
+    
+    // Check if we're already listing a directory
+    if (m_listState == LISTING_IN_PROGRESS) {
+        this->log_WARNING_HI_ListDirectoryError(logStringDirName, static_cast<U32>(Os::Directory::OTHER_ERROR));
+        this->sendCommandResponse(opCode, cmdSeq, Os::FileSystem::OTHER_ERROR);
+        return;
+    }
+    
+    this->log_ACTIVITY_HI_ListDirectoryStarted(logStringDirName);
+
+    // Open the directory for reading
+    Os::Directory::Status status = m_currentDir.open(dirName.toChar(), Os::Directory::OpenMode::READ);
+    
+    if (status != Os::Directory::OP_OK) {
+        this->log_WARNING_HI_ListDirectoryError(logStringDirName, static_cast<U32>(status));
+        this->emitTelemetry(Os::FileSystem::OTHER_ERROR);
+        this->sendCommandResponse(opCode, cmdSeq, Os::FileSystem::OTHER_ERROR);
+        return;
+    }
+
+    // Initialize state machine for asynchronous processing
+    m_listState = LISTING_IN_PROGRESS;
+    m_currentDirName = dirName.toChar();
+    m_currentOpCode = opCode;
+    m_currentCmdSeq = cmdSeq;
+    m_currentIndex = 0;
+    m_totalEntries = 0;
+    
+    // Directory listing will be processed asynchronously by Rate Group 2 (0.5Hz).
```

Comment:
RGs are configurable per project.  Remove the `2 and (0.5Hz)` or note it as an example.